totalRP3/Modules/NamePlates/NamePlates_Plater.lua

```diff
@@ -0,0 +1,90 @@
+
+local TRP3_PlaterNamePlates = {} -- magic public table
+
+function TRP3_PlaterNamePlates:OnModuleInitialize()
+	if GetAddOnEnableState(nil, "Plater") ~= 2 then
+		return false, L.NAMEPLATES_MODULE_DISABLED_BY_DEPENDENCY;
+	elseif TRP3_NAMEPLATES_ADDON ~= nil then
+		return false, L.NAMEPLATES_MODULE_DISABLED_BY_EXTERNAL;
+	end
+
+	if self.PlaterMod then
+		Plater.RecompileScript(self.PlaterMod);
+	end
+end
+
+function TRP3_PlaterNamePlates:OnModuleEnable()
+	if not IsAddOnLoaded("Plater") then
+		return false, L.NAMEPLATES_MODULE_DISABLED_BY_DEPENDENCY;
+	end
+
+	Plater = _G.Plater;
+
+	-- Setting the global TRP3_NAMEPLATES_ADDON is done via the mod and should not be set here.
+	local success, scriptAdded, _ = Plater.ImportScriptString(importString, false);
+
+	-- If the mod was not installed (is already up to date) then find the installed mod object so we can still control it
+	if not success and not scriptAdded then
+		local scriptType = "hook";
+
+		local scriptDB = Plater.GetScriptDB(scriptType);
+		local newScriptTable = Plater.DecompressData(importString, "print");
+		local newScript = Plater.BuildScriptObjectFromIndexTable(newScriptTable, scriptType);
+
+		for i = 1, #scriptDB do
+			local scriptObject = scriptDB[i];
+			if scriptObject.Name == newScript.Name then
+				scriptAdded = scriptObject;
+			end
+		end
+	end
+
+	if not scriptAdded then return false, L.PLATER_NAMEPLATES_WARN_MOD_IMPORT_ERROR end
+
+	scriptAdded.Enabled = true;
+
+	Plater.RecompileScript(scriptAdded);
+
+	TRP3_NAMEPLATES_ADDON = "Plater";
+
+	Plater.ForceTickOnAllNameplates();
+
+	self.PlaterMod = scriptAdded;
+end
+
+function TRP3_PlaterNamePlates:OnModuleDisable()
+	self.PlaterMod.Enabled = false;
+
+	if not TRP3_PlaterNamePlates.PlaterMod.Enabled then
+		TRP3_NAMEPLATES_ADDON = nil;
```

Comment:
As hot-reloading is new - what would happen in the following circumstance:

1. I log in with Kui nameplates installed and enabled.
2. TRP initializes the Kui nameplates decorator, sets `TRP3_NAMEPLATES_ADDON` to "KuiNamePlates" (or whatever it was).
3. I enter the TRP settings and disable the Plater module.

In this case I'd expect that `TRP3_NAMEPLATES_ADDON` would inappropriately get unset despite having been activated for an entirely unrelated addon.

If so - it'd be wise to gate this instead on `if TRP3_NAMEPLATES_ADDON == "Plater"` I'd think.

Reply:
I'll look into testing this more thoroughly next week but on my first set of mental gymnastics, I do think this is dangerous and will need to be looked at, will probably just implement your suggestion. It will need some additional work in there somewhere though as the mod will reset the flag to nil when it's disabled, independently of the TRP module.

tldr: needs fix and will fix next week

Reply:
The modules would not be able to properly initialize if another one got to the bag first. In the module initialization function for both of them they check if `TRP3_NAMEPLATES_ADDON` is set and will disable themselves if it's not `nil`.

This means that while the scenario you mentioned _should_ be impossible, I am still going to implement your suggestion just to be safe.